### ISSUE TITLE
Warn if no credentials are in effect for S3

### ISSUE THREAD

**Issue by qc00** [man-group/ArcticDB#473]
**Is your feature request related to a problem? Please describe.**
When setting S3 credentials via environment variables, `aws_auth=true` also has to be set, but there's no explicit information about this.

**Describe the solution you'd like**
Possibilities:
* Require the URI to have a credential (aws_auth=true or credentials in URI)
* Print a warning if `AWS_*` environment variables are set but `aws_auth` is not.

**Comment by mehertz** [comment]
I've been thinking about this.

> Print a warning if AWS_* environment variables are set but aws_auth is not.

They might not be using AWS env vars. I think this is quite hard to do actually.

Maybe we should focus our efforts on standardising the URI instead?

**Comment by qc00** [comment]
Yeah, I don't want to tightly couple to any AWSSDK-specific stuff, either.

Maybe something like: if no explicit credentials or `aws_auth`, make a test connection to the bucket.

> Maybe we should focus our efforts on standardising the URI instead?

I am not sure that would necessarily help. My understanding of the user's issue is: the library appears to "open" successfully but the first call fails with an opaque error.

**Comment by mehertz** [comment]
> if no explicit credentials or aws_auth, make a test connection to the bucket.

I think that's a good idea. Maybe just HEAD the bucket? Hope that would give a non-200 if you don't have any creds.

**Comment by mehertz** [comment]
We need to be careful not to let this overload storages though. Would be adding a new request per object creation.
